src/utils.ts

```diff
@@ -179,27 +176,53 @@ export function tokenToPythSymbol(token: string) {
   return undefined;
 }
 
-export async function getPythPrice(splMint: PublicKey) {
+export async function getPythPrice(baseMint: PublicKey) {
   const connection: Connection = new Connection(API_URL);
   const pythPublicKey = getPythProgramKeyForCluster(IS_DEV ? 'devnet': 'mainnet-beta');
   const pythClient = new PythHttpClient(connection, pythPublicKey);
   const data = await pythClient.getData();
 
   for (let symbol of data.symbols) {
     const price = data.productPrice.get(symbol)!;
-    if (tokenToPythSymbol(splMintToToken(splMint)) == symbol) {
+    if (tokenToPythSymbol(splMintToToken(baseMint)) == symbol) {
       return price.price;
     }
   }
   return 0.0;
 }
 
-export function tokenType(type: string) {
-  if (type == "Upside") {
-    return "call";
+export function getDIPType(dip_deposit: DIPDeposit) {
+  for (let premium of premiumAssets){
```

Comment:
using a for loop to iterate over premium will work, but is unnecessary. Fine for now, but something to clean up later.